### DISCUSSION TITLE
Typescript Enum is not assignable to Prisma Enum

### DISCUSSION THREAD

**Discussion by fnoquiq** [prisma/prisma#9215] [Q&A] [answered]
my schema
```prisma
enum Role {
  USER
  ADMIN
}
```

my Role domain
```ts


enum Role {
  USER,
  ADMIN,
}

export { Role }
```

My prisma repository
```ts
import { Role as PrismaRole } from '.prisma/client'
import { Role as DomainRole } from '../domain/role'

const test: PrismaRole = Role.ADMIN 
ERROR // Type 'Role.ADMIN' is not assignable to type 'Role'
```

**Comment by ryands17** [comment]
@fnoquiq 👋 

The enum created by Prisma is not actually a type but a string literal. You would need to import the type defined by Prisma like this:

```ts
import type { Role as PrismaRole } from '.prisma/client'

const test: PrismaRole = Role.ADMIN
```

**Reply by fnoquiq** [reply]
But I wouldn't want to import anything from the prisma. My domain layer cannot be dependent on the prisma. 
I would like to try to reproduce the type (Role) generated by the prisma

Something like
```ts
export enum MyRoleType {
 USER,
 ADMIN,
}

let myRoleType: MyRoleType
myRoleType = PrismaRole.ADMIN
// but MyRoleType isn`t compatible with PrismaRole
```

**Comment by fnoquiq** [comment, marked as answer]
This works!!

```ts
export const Role: {
  USER: 'USER'
  ADMIN: 'ADMIN'
} = {
  USER: 'USER',
  ADMIN: 'ADMIN',
}

export type Role = typeof Role[keyof typeof Role]
```

**Reply by MattHalloran** [reply]
This slightly condensed version is also working for me:
```typescript
export const Role: { [x: string]: 'USER' | 'ADMIN'} = {
  USER: 'USER',
  ADMIN: 'ADMIN',
}

export type Role = typeof Role[keyof typeof Role]
```

**Reply by viniciusdeliz** [reply]
Thank you. Is exporting the `const` necessary though? Isn't only the type enough?

**Reply by achtan** [reply]
This works as well
```ts
export const ROLE = {
  USER: 'USER',
  ADMIN: 'ADMIN',
} as const

export type Role = typeof Role[keyof typeof Role]
```

**Reply by woss** [reply]
Does this work only on the frontend or in nodejs?

**Reply by JbIPS** [reply]
Giving a value to your enum members seems to works just fine:
```ts
enum Role {
  USER = 'USER',
  ADMIN = 'ADMIN',
}
```
is accepted on my prisma client methods

**Comment by aravindvakil** [comment]
Enums can be used in TS interfaces by importing from the Prisma client library.

`import { roleEnum } from '@prisma/client'
`

We can also create an enum as a type, But the enum should match the Prisma client type.

`type Role = "admin" | "user"
`

**Reply by ufb007** [reply]
This one worked for me perfectly. Thanks buddy.

**Reply by woss** [reply]
this approach doesn't work when used in frontend

**Reply by fadexadex** [reply]
Thank man, this worked for me.
